controllers/helpers.go

```diff
@@ -98,3 +98,9 @@ func secretKeyRef(ctx context.Context, c client.Client, nn types.NamespacedName,
 
 	return "", fmt.Errorf("unable to find key=%q in secret=%q namespace=%q", key, nn.Name, nn.Namespace)
 }
+
+// truncateLabelValue truncates the given string to 63 characters, which is the max length
+// of a Kubernetes label value.
+func truncateLabelValue(label string) string {
+	return label[:min(len(label), 63)]
```

Comment:
It's possible that this can generate an invalid label that doesn't match the requirements - the example I saw was when the value gets truncated and ends with a `-`

> a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')"}